### PR TITLE
Add discount properties

### PR DESCRIPTION
**Pull request by jenniferboedker**
This PR adds the totalPrice and discountPrice to the productitem and the totalDiscount price to the offer DTO

### CODE REVIEW

**Comment by jenniferboedker** [comment]
[#686: adjust DTOs in DML to represent the discount in the product items](https://trello.com/c/SpHnrSo1/893-686-adjust-dtos-in-dml-to-represent-the-discount-in-the-product-items)

File: src/main/groovy/life/qbic/datamodel/dtos/business/ProductItem.groovy [hunk 1]
```diff
@@ -23,10 +23,39 @@ class ProductItem {
...
 
+    /**
+     * Describes the total price of an item based on the quantity and unit price of a product
+     */
+    final double totalPrice
+
+    /**
+     * Describe the total discount price for a product (based on the quantity)
+     */
+    final double totalDiscountPrice
```
Comment (sven1103) [inline src/main/groovy/life/qbic/datamodel/dtos/business/ProductItem.groovy:34]: The commercial term is `quantity discount`, and should be the financial volume difference to the total price.
```suggestion
    final double quantityDiscount
```

File: src/main/groovy/life/qbic/datamodel/dtos/business/ProductItem.groovy [hunk 2]
```diff
@@ -23,10 +23,39 @@ class ProductItem {
...
+     * @param product The product for which an item is created
+     *
+     * @deprecated 2.11.0
+     */
+    @Deprecated
     ProductItem(double quantity, Product product) {
         this.quantity = quantity
         this.product = product
+        this.totalPrice = 0
+        this.totalDiscountPrice = 0
```
Comment (sven1103) [inline src/main/groovy/life/qbic/datamodel/dtos/business/ProductItem.groovy:48]: renaming suggestion as above

File: src/main/groovy/life/qbic/datamodel/dtos/business/ProductItem.groovy [hunk 3]
```diff
@@ -23,10 +23,39 @@ class ProductItem {
...
     }
 
+    ProductItem(double quantity, Product product, double totalPrice, double totalDiscountPrice) {
+        this.quantity = quantity
+        this.product = product
+        this.totalPrice = totalPrice
+        this.totalDiscountPrice = totalDiscountPrice
```
Comment (sven1103) [inline src/main/groovy/life/qbic/datamodel/dtos/business/ProductItem.groovy:55]: same :)

**Review by sven1103 — CHANGES_REQUESTED** [review]
Thanks a lot @jenniferboedker, I made a small comment to apply the correct financial term

File: src/main/groovy/life/qbic/datamodel/dtos/business/Offer.groovy [hunk 4]
```diff
@@ -132,6 +136,7 @@ class Offer {
          */
         double totalPrice
         double netPrice
+        double totalDiscountPrice
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/Offer.groovy:139]: the official term would be `amount` for each price. So here I would suggest `discountAmount`
https://www.invoiceberry.com/free-pdf-invoice-templates

File: src/main/groovy/life/qbic/datamodel/dtos/business/ProductItem.groovy (hunk 1 above)
Comment (jenniferboedker) [reply src/main/groovy/life/qbic/datamodel/dtos/business/ProductItem.groovy:34]: Can we do quanityDiscountPrice? So its clear its a price and not the ratio? @sven1103

File: src/main/groovy/life/qbic/datamodel/dtos/business/Offer.groovy (hunk 4 above)
Comment (jenniferboedker) [reply src/main/groovy/life/qbic/datamodel/dtos/business/Offer.groovy:139]: But then totalPrice would also not be correct, right

Comment (sven1103) [reply src/main/groovy/life/qbic/datamodel/dtos/business/Offer.groovy:139]: What about `appliedTotalQuantityDiscount`?

Comment (jenniferboedker) [reply src/main/groovy/life/qbic/datamodel/dtos/business/Offer.groovy:139]: I would stay with the naming scheme we already have. At least for me its clear what it means. If we want to stick to a naming schema (not include the price) we should adapt this for the other variables as well (totalPrice, netPrice,..)

**Review by sven1103 — APPROVED** [review]

**Pull request merged**